### DISCUSSION TITLE
Custom hooks to hang HTML onto

### DISCUSSION THREAD

**Discussion by example-user30** [GetPublii/Publii#1394] [Ideas]
I would like to be able to create some HTML sections that can be inserted into a Post and rendered when the project is previewed or published.  
There are several ways to do this:  
1. Maybe a shortcode editor that is like the HTML editor accessed from the Tools & Plugins.  When a shortcode is created it makes an individual HTML file that is assigned that shortcode name.  Edit the HTML file (provided CSS support with a CSS editor as well would be neat).  Then in the block editor or  WYSUWYG editor you could add a shortcode and pick from existing shortcodes.
2. The other way to do this would be to define a hook from inside the editor (either block or WYSIWYG) that you would give a specific name (reusing names should be possible).  Then in the Custom HTML editor add an extra button for each hook created.  In this case the CSS would need to be created in the Custom CSS section.

![html editor](https://github.com/GetPublii/Publii/assets/156127535/cd0efb4f-54fc-46a4-b380-92e632d6aa22)

Thanks

**Comment by dziudek** [comment]
Hi,

You can create "shortcodes" using the Smart Replacer plugin: https://getpublii.com/docs/smart-replacer.html - it will works inside posts, but you can make it working in any place of your website

**Reply by example-user30** [reply]
I have not purchased that yet.  I will look into it and see if it meets the requirement.  Thanks

**Comment by lukajincharadze** [comment]
`Both approaches you've outlined could work well for creating HTML sections to be inserted into a post and rendered when previewed or published. Let's discuss each option:

Shortcode Editor:

This approach involves creating a shortcode editor where users can create and edit individual HTML files assigned to shortcode names.
Users would access the shortcode editor, create a new shortcode, and then edit the corresponding HTML file, potentially with CSS support.
In the block editor or WYSIWYG editor, users could then insert a shortcode and select from existing shortcodes.
Pros:
Provides a dedicated interface for managing shortcodes and their corresponding HTML content.
Allows for easy reuse of existing shortcodes.
Cons:
Requires managing multiple HTML files, which could become cumbersome with a large number of shortcodes.
May require additional development effort to implement the shortcode editor and integrate it with the block editor or WYSIWYG editor.
Defined Hooks and Custom HTML Editor:

In this approach, users define hooks from within the block or WYSIWYG editor and then add corresponding HTML content in the Custom HTML editor.
Each hook would have a specific name, and users could reuse hook names if needed.
The Custom HTML editor would include an extra button for each defined hook, allowing users to easily insert the corresponding HTML content.
Pros:
Integrates directly with existing WordPress functionality, as users are already familiar with the block editor and Custom HTML editor.
Simplifies the management of HTML content by centralizing it in the Custom HTML editor.
Cons:
Users may need to manually manage CSS in the Custom CSS section, potentially leading to inconsistencies or conflicts.
Limited flexibility compared to a dedicated shortcode editor in terms of managing and organizing HTML content.
Ultimately, the choice between these approaches depends on factors such as user preferences, the complexity of the project, and the desired level of integration with existing WordPress functionality. Both options have their merits and could be viable solutions depending on your specific requirements and constraints.`

**Reply by example-user30** [reply]
I really appreciate your adding to this suggestion.  This is kind of how I saw this as well.  

You kept calling this WordPress, but it is Publii.  Right?

Thanks
